sdk/resourcemanager/Azure.ResourceManager/src/Extensions/ArmClientBuilderExtensions.cs

```diff
@@ -23,6 +24,8 @@ public static IAzureClientBuilder<ArmClient, ArmClientOptions> AddArmClient<TBui
         /// <summary>
         /// Registers an <see cref="ArmClient"/> instance with connection options loaded from the provided <paramref name="configuration"/> instance.
         /// </summary>
+        [RequiresUnreferencedCode("Uses reflection for registration")]
+        [RequiresDynamicCode("Uses code generation for registration")]
```

Comment:
nit - The message could be more descriptive like what I have in this one [here](https://github.com/m-redding/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/api/Azure.Core.net8.0.cs#L20)"This utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming." 
idk if there's an AOT safe alternative path here but if there is adding that would also be a plus